src/aabb.rs

```diff
@@ -0,0 +1,76 @@
+use crate::vec::V3;
+use std::ops::Add;
+use crate::ray::Ray;
+
+#[derive(Copy, Clone, Debug)]
+pub struct AABB {
+    min: V3,
+    max: V3,
+}
+
+impl AABB {
+    pub fn new(min: V3, max: V3) -> AABB {
+        AABB { min, max }
+    }
+
+    pub fn hit(self, ray: &Ray, d_min: f64, d_max: f64) -> bool {
+         {
+            let inv_d = 1.0 / ray.direction().x;
+            let start = ray.origin().x;
+            let mut d0 = (self.min.x - start) * inv_d;
+            let mut d1 = (self.max.x - start) * inv_d;
+            if inv_d.is_sign_negative() {
+                std::mem::swap(&mut d0, &mut d1);
+            }
+            let min = if d_min < d0 { d0 } else { d_min };
```

Comment:
I believe that `f64::max` can be used here.

Reply:
It performs additional checks (NaN/Inf related) and requires 4 AVX instructions instead of 1, left as is, since it's very hot method.